sw/airborne/modules/relativelocalizationfilter/relativelocalizationfilter.c

```diff
@@ -0,0 +1,203 @@
+		keepbounded(&ownVx,-2.0,2.0);
+		keepbounded(&ownVy,-2.0,2.0);
+
+		// Make the filter only in Guided mode (flight).
+		// This is because it is best for the filter should only start once the drones are in motion, 
+		// otherwise it might diverge while drones are not moving.
+		if (guidance_h.mode == GUIDANCE_H_MODE_GUIDED)
+		{
+			ekf[i].dt = (get_sys_time_usec() - now_ts[i])/pow(10,6); // Update the time between messages
+			
+			// Get the velocity in NED for the tracked aircraft
+			float trackedVx, trackedVy;
+			polar2cart(acInfoGetGspeed(ac_id), acInfoGetCourse(ac_id), &trackedVx, &trackedVy); // get North and East velocities (m/s)
+			// As for own velocity, bind to realistic amounts to avoid occasional spikes/NaN/inf errors
+			keepbounded(&trackedVx,-2.0,2.0);
+			keepbounded(&trackedVy,-2.0,2.0);
+
+			// Construct measurement vector Y for EKF using the latest data obtained.
+			// Y = [RSSI owvVx ownVy trackedVx trackedVy dh], EKF_M = 6 as defined in discreteekf.h
+			float Y[EKF_M];
+			Y[0] = (float)rssi; 	//RSSI measurement
+			Y[1] = ownVx; 	   		// Own velocity North (NED frame)
+			Y[2] = ownVy;			// Own velocity East  (NED frame)
+			Y[3] = trackedVx;  		// Velocity of other drone Norht (NED frame)
+			Y[4] = trackedVy;		// Velocity of other drone East  (NED frame)
+			Y[5] = acInfoGetPositionUtm_f(ac_id)->alt - stateGetPositionEnu_f()->z;  // Height difference
+			
+			// Run the steps of the EKF, but only if velocity difference is significant (to filter out minimal noise)
+			if (  sqrt( pow(Y[1]-Y[3],2) + pow(Y[2]-Y[4],2) ) > 0.05 )
+			{
+				ekf_filter_predict(&ekf[i], &model[i]); // Prediction step of the EKF 
+				ekf_filter_update(&ekf[i], Y);	// Update step of the EKF
+			}
+		}
+
+		now_ts[i] = get_sys_time_usec();  // Store latest time
+
+	}
+};
+
+#ifdef PPRZ_MSG_ID_RLFILTER
+int cnt;
+static void send_rafilterdata(struct transport_tx *trans, struct link_device *dev)
+{
+	// To avoid overflowing, it is best to send the data of each tracked drone separately.
+	// To do so, we can cycle through the filters at each new timestep.
+	cnt++;
+	if (cnt == nf)
+		cnt = 0;
+
+	pprz_msg_send_RLFILTER(
+		trans, dev, AC_ID,			 // Standard stuff
+		&id,			     		 // ID of the tracked UAV in question
+		&RSSIarray[i], 		    	 // Received ID and RSSI
+		&srcstrength[i],		     // Source strength
+		&ekf[i].X[0], &ekf[i].X[1],  // Relative position [North, East]
+		&ekf[i].X[2], &ekf[i].X[3],  // Own velocity [North, East]
+		&ekf[i].X[4], &ekf[i].X[5],  // Relative velocity of other drone [North, East]
+		&ekf[i].X[6]				 // Height separation [Down]
+		); 
+			 
+};
+#endif
+
+void relativelocalizationfilter_init(void)
+{
+	array_make_zeros_int(NUAVS-1, IDarray); // Clear out the known IDs
+	nf = 0; // Number of active filters upon initialization
+
+	AbiBindMsgRSSI(ABI_BROADCAST, &rssi_ev, bluetoothmsg_cb); // Subscribe to the ABI RSSI messages
+	#ifdef PPRZ_MSG_ID_RLFILTER
```

Comment:
This define doesn't make much sens. If the message is available in PPRZLINK, it will be defined, and this pull request should include this otherwise it doesn't make much sense. On the other hand, the register function are usually protected by `#if PERIODIC_TELEMETRY` like here: https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/subsystems/gps.c#L340